bofire/data_models/objectives/desirabilities.py

```diff
@@ -0,0 +1,219 @@
+from typing import Optional, Union
+
+import numpy as np
+import pandas as pd
+import pydantic
+
+from bofire.data_models.objectives.identity import IdentityObjective
+
+
+class _SeriesNumpyCallable:
+    """Helper class to call numpy and torch functions with series or numpy arrays. matches __call__
+    signature of objectives."""
+
+    def __call__(
+        self, x: Union[pd.Series, np.ndarray], x_adapt
+    ) -> Union[pd.Series, np.ndarray]:
+        convert_to_series = False
+        if isinstance(x, pd.Series):
+            convert_to_series = True
+            name = x.name
+            x = x.values
+
+        y = self.call_numpy(x)
+
+        if convert_to_series:
+            return pd.Series(y, name=name)
+
+        return y
+
+    def call_numpy(self, x: np.ndarray) -> np.ndarray:
+        raise NotImplementedError()
+
+
+class _LogShapeClipFactorValidator:
+    @pydantic.field_validator("clip", check_fields=False)
```

Comment:
For what is the `check_fields=False` needed?

Reply:
In general, it is recommended to use the `model_validator` with `mode="after"` if one wants to access also oter fields. Why to you factor out this functionality in a seperate class and not just include it directly into the specific data models? This would make the code more readable? What do you think?